src/app/clusters/water-heater-management-server/WaterHeaterManagementTestEventTriggerHandler.h

```diff
@@ -0,0 +1,86 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/TestEventTriggerDelegate.h>
+
+/**
+ * @brief User handler for handling the test event trigger
+ *
+ * @note If TestEventTrigger is enabled, it needs to be implemented in the app
+ *
+ * @param eventTrigger Event trigger to handle
+ *
+ * @retval true on success
+ * @retval false if error happened
+ */
+bool HandleWaterHeaterManagementTestEventTrigger(uint64_t eventTrigger);
```

Comment:
This should NOT be global and required to implement via global linkage. Please add this to the delegate and/or instance, which are the ones which have all the state that will need to change on those test even triggers

Reply:
Happy to change this but I noticed from the slide deck you sent me a "good example" of TestEventTriggers is src/app/clusters/energy-evse-server/EnergyEvseTestEventTriggerHandler.h and I think WaterHeaterManagementTestEventTriggerHandler.h is replica of this.

`HandleWaterHeaterManagementTestEventTrigger `is only actually needed when in connectedhomeip/examples/platform/linux/AppMain.cpp, `CHIP_DEVICE_CONFIG_ENABLE_WATER_HEATER_MANAGEMENT_TRIGGER `is defined and `WaterHeaterManagementTestEventTriggerHandler `is instantiated.

Do you have an example of the way you suggested above please?

Reply:
OK/thanks

Reply:
Marking as resolved.